kubetest/anywhere.go

```diff
@@ -317,6 +317,65 @@ func (_ *kubernetesAnywhere) KubectlCommand() (*exec.Cmd, error) { return nil, n
 
 const defaultConfigFile = ".config"
 
+/*
+ multiClusterDeployment type holds the data passed to `--multi-clusters` flag.
+ The format of value that should be passed to the flag is `[Zone1:]Cluster1[,[ZoneN:]ClusterN]]*`.
+ Multiple clusters can be specified as a comma separated list.
+ Zone can be optionally specified along with cluster name as described above in the format.
+ If zone is not specified along with cluster then cluster would be deployed in default zone.
+*/
+type multiClusterDeployment struct {
+	zones    map[string]string
+	clusters []string
+}
+
+func (m *multiClusterDeployment) String() string {
+	var str string
+	for _, cluster := range m.clusters {
+		if len(str) != 0 {
+			str += ","
+		}
+		zone, exist := m.zones[cluster]
+		if exist {
+			str += zone + ":"
+		}
+		str += cluster
+	}
+	return str
+}
+
+func (m *multiClusterDeployment) Set(value string) error {
+	if len(value) == 0 {
+		return errors.New("invalid value passed to --multi-clusters flag, should specify at least one cluster")
+	}
+
+	if m.zones == nil {
+		m.zones = make(map[string]string)
+	}
+	clusterZones := strings.Split(value, ",")
+	for _, czTuple := range clusterZones {
+		czSlice := strings.SplitN(czTuple, ":", 2)
+		if len(czSlice[0]) == 0 || (len(czSlice) == 2 && len(czSlice[1]) == 0) {
+			return errors.New("invalid value passed to --multi-clusters flag")
+		}
+		if len(czSlice) == 2 {
+			m.zones[czSlice[1]] = czSlice[0]
+			m.clusters = append(m.clusters, czSlice[1])
+		} else {
+			m.clusters = append(m.clusters, czSlice[0])
+		}
+	}
+	return nil
+}
+
+func (m *multiClusterDeployment) Type() string {
+	return "multiClusterDeployment"
+}
+
+func (m *multiClusterDeployment) Enabled() bool {
+	return len(m.clusters) > 0
+}
+
```

Comment:
cc @neolit123 I dumped this struct from federation.go into k-a.go, although seems there's no CI job uses this, let me know if you prefer just get rid of that :-)

Reply:
@krzyzacy 
+1 for removing these multicluster bits.
i'm hoping that we can remove `anywhere.go` as a whole in a couple of releases completely.

Reply:
thanks! cleaned up :-)